src/weather_station/connection.c

```diff
@@ -0,0 +1,198 @@
+#include <stdio.h>
```

Comment:
please write in all files what they are supposed to do. The name "connection" is also very general.

Reply:
I didn't perform that task because the C files were intended for testing the connection to the slaves. Therefore, I don't see any reason to do that right now.

Reply:
Something like this is enough:

```suggestion
/// Test connection to Thies Weather station by reading a register

#include <stdio.h>
```